cachito/workers/scm.py

```diff
@@ -0,0 +1,201 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import os
+from abc import ABC, abstractmethod
+import urllib.parse
+import shutil
+import tarfile
+import tempfile
+import subprocess
+
+import requests
+
+from cachito.errors import ConfigError, CachitoError
+
+
+log = logging.getLogger(__name__)
+
+
+# TODO: Write tests
+class SCM(ABC):
+    """The base class for interacting with source control."""
+
+    def __init__(self, url, ref):
+        """
+        Initialize the SCM class.
+
+        :param str url: the source control URL to the repository to fetch
+        :param str ref: the source control reference
+        """
+        super().__init__()
+        self.url = url
+        self.ref = ref
+        self._archives_dir = None
+        self._archive_path = None
+        self._repo_name = None
+        # Validate that the base directory can be retrieved
+        self.archives_dir
```

Comment:
It sounds like you don't want `self.archives_dir` to be a property :)
Maybe just have a helper method that is called here to do the work?

Reply:
I'll keep it as a property and move the validation to `validate_celery_config` so this line can be removed.